pkg/mediationcontainer/sdk_client_protocol.go

```diff
@@ -12,21 +13,42 @@ import (
 )
 
 const (
-	waitResponseTimeOut = time.Second * 30
+	waitRegistrationResponseTimeOut = time.Second * 300
+	waitResponseTimeOut             = time.Second * 300
 )
 
 type SdkClientProtocol struct {
-	allProbes                   map[string]*ProbeProperties
-	version                     string
-	communicationBindingChannel string
+	allProbes                              map[string]*ProbeProperties
+	version                                string
+	communicationBindingChannel            string
+	waitRegistrationResponseTimeOut        time.Duration
+	exitProbeOnRegistrationResponseTimeOut bool
 	//TransportReady chan bool
 }
 
-func CreateSdkClientProtocolHandler(allProbes map[string]*ProbeProperties, version, communicationBindingChannel string) *SdkClientProtocol {
+func CreateSdkClientProtocolHandler(allProbes map[string]*ProbeProperties, version, communicationBindingChannel string,
+	sdkProtocolConfig *SdkProtocolConfig) *SdkClientProtocol {
+	var defaultResponseTimeOut time.Duration
+	var exitProbeOnRegistrationResponseTimeOut bool
+
+	if sdkProtocolConfig == nil {
+		defaultResponseTimeOut = waitRegistrationResponseTimeOut
+		exitProbeOnRegistrationResponseTimeOut = false
+	} else {
+		var timeout time.Duration
+		timeout = time.Duration(sdkProtocolConfig.RegistrationTimeoutSec)
```

Comment:
Shall we do some basic sanity checks here? For example, the value should not be smaller than the default value? Or at least not smaller than 60 seconds?

Reply:
Added check to default value to 300 secs if the provided value is less than 60 secs